File: lib-segment/src/main/java/websiteschema/mpsegment/hmm/TrieNodeBinarySort.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package websiteschema.mpsegment.hmm;

import java.util.*;

/**
 *
 * @author ray
 */
public class TrieNodeBinarySort implements TrieNodeSortor {

    Comparator<Trie> comparator = null;
    
    public TrieNodeBinarySort() {
        comparator = new Comparator<Trie>() {

            @Override
            public int compare(Trie o1, Trie o2) {
                return o1.key - o2.key;
            }
        };
    }

    @Override
    public Trie[] sort(Trie[] values) {
        List<Trie> tmp = new ArrayList<Trie>();
        tmp.addAll(Arrays.asList(values));
        Collections.sort(tmp, comparator);
        return tmp.toArray(values);
    }
}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/hmm/TrieNodeBinarySort.scala
package websiteschema.mpsegment.hmm

class TrieNodeBinarySort extends TrieNodeSortor {

    override def sort(values: Array[Trie]) : Array[Trie] = {
        val tmp = values.toList.sortBy(t => t.key)
        return tmp.toArray;
    }
}
